app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -284,59 +305,103 @@ export function computeTxPlan(
   const minimalChangeLovelace = computeMinUTxOLovelaceAmount(change.tokenBundle)
 
   // if we cannot create a change output with minimal ada we add it to the fee
-  if (change.tokenBundle.length === 0 && change.coins < minimalChangeLovelace) {
+  if (
+    allowAddingChangeToFee &&
+    change.tokenBundle.length === 0 &&
+    change.coins < minimalChangeLovelace
+  ) {
     return {
+      success: true,
+      txPlan: {
+        inputs,
+        outputs,
+        change: null,
+        certificates,
+        deposit,
+        additionalLovelaceAmount,
+        fee: (feeWithChange + change.coins) as Lovelace,
```

Comment:
shouldn't this be "fee without change"?

Reply:
Thnx, it seems it should :+1:

Reply:
Interesting, I couldnt wrap my head around how it previously worked if it is wrong, but it turned out to be correct.
`(feeWithoutChange + (change.coins + (feeWithChange - feeWithoutChange)))` which is basically same as `feeWithChange + change.coins`
We can discuss this in slack but maybe you are already getting that the change amount is firstly calculated like
`totalInput - totalOutput - feeWithChange` so for this special condition we need to add the difference between `feeWithoutChange` and `feeWithChange` to the change.

Reply:
I see, basically you need to discard the change output ADA towards the fee that was computed for the transaction containing that output - rather than putting a long expressions that is obviously simplifiable I'd put a comment explaining that - that's more valuable for somebody trying to understand what's going on